packages/app/src/systems/Account/hooks/useEns.ts

```diff
@@ -0,0 +1,72 @@
+import { ethers } from 'ethers';
+import { useEffect, useState } from 'react';
+import { isValidEthAddress } from '~/systems/Core';
+
+export type EnsData = {
+  name: string | null;
+  avatar: string | null;
+  loading: boolean;
+  error: Error | null;
+};
+
+export function useEns(address?: string) {
+  const [ensData, setEnsData] = useState<EnsData>({
+    name: null,
+    avatar: null,
+    loading: false,
+    error: null,
+  });
+
+  useEffect(() => {
+    async function resolveEns() {
+      if (!address || !isValidEthAddress(address)) {
+        setEnsData({
+          name: null,
+          avatar: null,
+          loading: false,
+          error: null,
+        });
+        return;
+      }
+
+      setEnsData((prev) => ({ ...prev, loading: true, error: null }));
+
+      try {
+        // Using Ethereum mainnet for ENS resolution
+        const provider = new ethers.providers.JsonRpcProvider(
+          'https://mainnet.infura.io/v3/84842078b09946638c03157f83405213'
```

Comment:
Maybe something like this could be done for the URL
https://github.com/FuelLabs/fuel-explorer-private/blob/feb70258a784e390f5d61f2532666c3fc6fc5322/packages/app-commons/src/config.ts